doc/deprecation-process.md

```diff
@@ -0,0 +1,272 @@
+# Rust Crate Deprecation Process
+
+## Overview
+
+This guide describes the step-by-step process for deprecating a Rust crate in the `azure-sdk-for-rust` repository. You likely need to read this if you are a crate maintainer and need to explain to your users that the crate should no longer be used.
+
+Note that a deprecated crate is a signal to users that they are strongly encouraged to stop using it and migrate to another crate. The crate is still available to install from crates.io and is not yanked, and we still have the ability to publish critical security fixes as necessary.
+
+The overall approach for Rust crates is:
+
+-   Add deprecation attributes to public APIs in the code to provide compile-time warnings
+-   Update the README.md file with a deprecation disclaimer
+-   Update the CHANGELOG.md file to document the deprecation
+-   Add deprecation metadata to Cargo.toml
+-   Publish a new release to crates.io
+-   Update the API reference documentation to show the deprecated status
+-   Eventually archive the crate on crates.io if the service is fully retired
```

Comment:
There's no archive functionality. We should be concrete in our recommendation. They should yank all versions, say.